backend/src/main/java/com/empire/World.java

```diff
@@ -2064,6 +2047,66 @@ void eatFood() {
 		}
 
 		void growPopulations() {
+			// Emigration
+			class Emigration {
+				final Region source;
+				final Region sink;
+				final double population;
+				public Emigration(Region source, Region sink, double population) {
+					this.source = source;
+					this.sink = sink;
+					this.population = population;
+				}
+			}
+			List<Emigration> emigrations = new ArrayList<>();
+			for (Region r : regions) {
+				double eligibleToEmigrate = r.population * r.unrestPopular * Constants.emigrationFactor;
+				if (starvingRegions.contains(r)) eligibleToEmigrate *= (1 + Constants.emigrationStarvationMod);
+				eligibleToEmigrate = Math.min(eligibleToEmigrate, r.population - 1); // Emigrating the last person causes a divide by zero error.
+				List<Region> destinations = new ArrayList<Region>();
+				for (Region n : r.getNeighbors(World.this)) {
+					if (n.type == Region.Type.LAND) destinations.add(n);
+					else for (Region nn : n.getNeighbors(World.this)) {
+						if (nn != r && n.type == Region.Type.LAND) destinations.add(nn);
```

Comment:
This seems to imply that destinations can be a neighbor of a neighbor to the region r being considered, hence not necessarily adjacent.  Just checking that this is intentional.

Reply:
Yes, but only if r is a water region.

Reply:
Ah you're right, I didn't process that correctly.